### PR TITLE
[Fix] Repoint pool table to root query

### LINKED ISSUE #6109: Repoint Admin Pool Table query at root query
## ♻️ Debt/Refactor

We would like to start relying on relationship scoping and partial results from our API instead of complex queries to ensure proper authorization.  Our Admin pool table query relies on navigating through the role assignment grqph.  Let's repoint it a simple root-level pool query and rely on the partial results.

## ✅ Acceptance Criteria

- [ ] Admin pool table uses a simple root-level query with partial results
- [ ] Verify correct partial-results are returned from the query based on auth

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #6109 

## 👋 Introduction

Update the `PoolTable` to use the root pools query instead of querying through the users team roles.

### Question

All roles that can access the pools table have the `view-any-publishedPool` permission. This is a little odd since on the admin side, I feel like I would only be able to see pools I can edit. Is this an issue? Should we filter out non-team pools specifically on admin or is it fine if they see them and just can't edit them?

### Answer

We only want to show pools the user has access to for administration purposes. We are filtering the pools so that pool operators can only see those their team owns.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run build`
2. Login as a platform admin
3. Add a pool on a specific team 
4. Log back in as a pool operator that is not on the team created in step 3
5. Navigate to the pools table
6. Confirm the pool created in step 3 does not exist

### CODE REVIEW

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx [hunk 1]
```diff
@@ -23,11 +23,8 @@ import {
   Maybe,
   Pool,
   Scalars,
-  useGetMePoolsQuery,
-  RoleAssignment,
   LocalizedString,
   useAllPoolsQuery,
-  Team,
```
Comment (petertgiles) [inline apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:30]: I think the team column is broken now.
image

**Review by petertgiles — CHANGES_REQUESTED** [review]
Wow, so much cleaner!  One thing to check on, though.

File: apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx (hunk 1 above)
Comment (esizer) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:30]: Oooo, good catch! 🙇‍♀️  Fixed in [query for team display name on pools table](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7449/commits/9e4fa79521cf734a8ac44e5d8b484fbf4660b95c)

Comment (petertgiles) [reply apps/web/src/pages/Pools/IndexPoolPage/components/PoolTable.tsx:30]: Looks great!

**Review by petertgiles — APPROVED** [review]

**Pull request merged**